File: Pelixy_Backend/models/Pelicula.js
class Pelicula {
  constructor(id, titulo, descripcion, anio, duracion, director, clasificacion, idioma, url_portada, fecha_registro) {
    this.id = id;
    this.titulo = titulo;
    this.descripcion = descripcion;
    this.anio = anio;
    this.duracion = duracion;
    this.director = director;
    this.clasificacion = clasificacion;
    this.idioma = idioma;
    this.url_portada = url_portada;
    this.fecha_registro = fecha_registro;
  }
}

module.exports = Pelicula;

File: models/Pelicula.java
package com.sise.pelixy.models;

import java.io.Serializable;

public class Pelicula implements Serializable {

    private int id;
    private String titulo;
    private String descripcion;
    private int anio;
    private String duracion;
    private String director;
    private String clasificacion;
    private String idioma;
    private String url_portada;
    private float calificacion;

    public Pelicula() {
        // Constructor vacío requerido por Retrofit y Gson
    }

    // Getters
    public int getId() {
        return id;
    }

    public String getTitulo() {
        return titulo;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public int getAnio() {
        return anio;
    }

    public String getDuracion() {
        return duracion;
    }

    public String getDirector() {
        return director;
    }

    public String getClasificacion() {
        return clasificacion;
    }

    public String getIdioma() {
        return idioma;
    }

    public String getUrlPortada() {
        return url_portada;
    }

    public float getCalificacion() {
        return calificacion;
    }

    public boolean tieneImagen() {
        return url_portada != null && !url_portada.isEmpty();
    }

    public void setId(int id) { this.id = id; }
    public void setTitulo(String titulo) { this.titulo = titulo; }
    public void setUrlPortada(String urlPortada) { this.url_portada = urlPortada; }

    public String[] obtenerGeneros() {
        return clasificacion != null ? clasificacion.split(",") : new String[0];
    }
}